packages/bbui/src/Drawer/Drawer.svelte

```diff
@@ -21,11 +22,27 @@
     }
     visible = false
   }
+
+  const easeInOutQuad = x => {
+    return x < 0.5 ? 2 * x * x : 1 - Math.pow(-2 * x + 2, 2) / 2
```

Comment:
Could you explain what is going on here, just for my benefit haha

Reply:
Haha of course! The purpose for replacing the default svelte `slide` transition in the first place is because it has an overshoot built in (goes up too far then jumps down a bit) and I think it looks gimmicky, so I wanted to replace it with our own slide transition.

This is just a quadratic function to smooth the new slide in animation for the drawer. Svelte provides a param `t` for custom animations, which goes from 0 to 1. If we linearly interpolated the animation relative to `t`, we would have a completely linear and smooth animation. This doesn't actually look particularly natural though - which is why CSS offers things like `ease-in` or `ease-out`, which slowly accelerates or decelerates the animation.

Since we're using a JS transition we can't use CSS helpers like those, so this is just a quadratic maths functions to minic what `ease-in-out` does. Plotted on a graph it looks like this:
![image](https://user-images.githubusercontent.com/9075550/155173273-e2a6b3d1-db5c-47c9-8fd4-2482abe7f267.png)

It basically starts off slightly slower, speeds up in the middle, and then slows down again towards the end. This makes the transition feel much smoother.

Reply:
Haha this is great. A true perfectionist.